packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -37,24 +62,38 @@ export async function resolvePackageManager(): Promise<PackageManagerInfo | unde
       } as const)
     );
 
-    const files = await fs.readdir(workspace);
+    const getDirFilesSortedByModificationDate = async (dir: string) => {
+      const files = await fs.promises.readdir(dir);
+
+      return files
+        .map((fileName) => ({
+          name: fileName,
+          time: fs.statSync(`${dir}/${fileName}`).mtime.getTime(),
+        }))
+        .sort((a, b) => a.time - b.time)
+        .map((file) => file.name);
+    };
+
+    const files = await getDirFilesSortedByModificationDate(workspace);
+    const packageManagerCandidates = [];
     for (const file of files) {
-      const packageManager = lockFiles.get(file);
-      if (packageManager) {
-        return packageManager;
+      const manager = lockFiles.get(file);
+      if (manager) {
+        packageManagerCandidates.push(packageManager);
       }
     }
-    try {
-      const packageManager = require(path.join(workspace, "package.json")).packageManager;
 
-      if (packageManager) {
-        // e.g. yarn@3.6.4
-        const match = packageManager.match(/^([a-zA-Z]+)@/);
-        return match ? match[1] : "npm";
-      }
-    } catch (e) {
-      // there might be a problem while reading package.json in which case we default to npm
+    if (packageManagerCandidates.length > 1) {
+      Logger.warn(
+        "Your workspace contains multiple package manager lock files, it might cause wrong manager to be used"
+      );
     }
+
+    if (packageManagerCandidates) {
+      return packageManagerCandidates[0];
+    }
+
+    // when no package manager were detected we default to npm
     return "npm";
```

Comment:
That also could be a const in one shared places, for example if we decide to use something else as defualt in the future.

Reply:
good point